digid_eherkenning/saml2/eherkenning.py

```diff
@@ -32,6 +35,214 @@
+            # for why we need this option.
+            "disableSignatureWrappingProtection": True,
+            # For eHerkenning, if the Metadata file expires, we sent them an update. So
+            # there is no need for an expiry date.
+            "metadataValidUntil": "",
+            "metadataCacheDuration": "",
+            "requestedAuthnContextComparison": "minimum",
+            "requestedAuthnContext": [eherkenning_config.loa],
+        },
+        # Service Provider Data that we are deploying.
+        "sp": {
+            # Identifier of the SP entity  (must be a URI)
+            "entityId": eherkenning_config.entity_id,
+            # Specifies info about where and how the <AuthnResponse> message MUST be
+            # returned to the requester, in this case our SP.
+            "assertionConsumerService": {
+                "url": furl(
+                    eherkenning_config.base_url + reverse("eherkenning:acs")
+                ).url,
+                "binding": "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Artifact",
+            },
+            "attributeConsumingServices": [
+                {
+                    "index": eherkenning_config.eh_attribute_consuming_service_index,
+                    "serviceName": eherkenning_config.service_name,
+                    "serviceDescription": eherkenning_config.service_description,
+                    "requestedAttributes": [
+                        {
+                            "name": "urn:etoegang:DV:%(oin)s:services:%(index)s"
+                            % {
+                                "oin": eherkenning_config.oin,
+                                "index": eherkenning_config.eh_attribute_consuming_service_index,
+                            },
+                            "isRequired": False,
+                        }
+                    ],
+                    "language": "nl",
+                },
+                {
+                    "index": eherkenning_config.eidas_attribute_consuming_service_index,
+                    "serviceName": eherkenning_config.service_name + " (eIDAS)",
+                    "serviceDescription": eherkenning_config.service_description,
+                    "requestedAttributes": [
+                        {
+                            "name": "urn:etoegang:DV:%(oin)s:services:%(index)s"
+                            % {
+                                "oin": eherkenning_config.oin,
+                                "index": eherkenning_config.eidas_attribute_consuming_service_index,
+                            },
+                            "isRequired": False,
+                        }
+                    ],
+                    "language": "nl",
+                },
+            ],
+            "NameIDFormat": "urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified",
+            "x509cert": open(cert_file, "r").read(),
+            "privateKey": open(key_file, "r").read(),
```

Comment:
should use a context manager to ensure the file is closed after generating metadata